scripts/common/scr_param.pm.in

```diff
@@ -13,64 +13,51 @@ sub new
   my $self = {};
 
   $self->{prog} = 'scr_param';
-  $self->{usrconf} = undef;
-  $self->{sysconf} = undef;
-  $self->{compile} = undef;
-  $self->{no_user} = undef;
+  $self->{usrconf} = {};
+  $self->{sysconf} = {};
 
-  # get current working dir
-  my $prefix = `pwd`;
-  chomp $prefix;
+  # set our compile time constants
+  $self->{compile} = {};
+  $self->{compile}{"CNTLDIR"}{"@SCR_CNTL_BASE@"} = {};
+  $self->{compile}{"CACHEDIR"}{"@SCR_CACHE_BASE@"} = {};
+  $self->{compile}{"SCR_CNTL_BASE"}{"@SCR_CNTL_BASE@"} = {};
+  $self->{compile}{"SCR_CACHE_BASE"}{"@SCR_CACHE_BASE@"} = {};
+  $self->{compile}{"SCR_CACHE_SIZE"}{"1"} = {};
 
-  # use value in $SCR_PREFIX if set
-  if (defined $ENV{SCR_PREFIX}) {
-    $prefix = $ENV{SCR_PREFIX};
-  }
-  
-  # tack on current working dir if needed
-  # don't resolve symlinks
-  # don't worry about missing parts, the calling script calling might create it
-  $prefix = `realpath -m -s $prefix`;
-  chomp $prefix;
+  # set our restricted parameters,
+  # these can't be set via env vars or user conf file
+  $self->{no_user} = {};
+  $self->{no_user}{"SCR_CNTL_BASE"} = {};
```

Comment:
We've switched SCR_CNTL_BASE to allow users to set it now.

Reply:
Fixed in d4b8fac